lib/chef/knife/core/subcommand_loader.rb

```diff
@@ -128,95 +143,65 @@ def find_subcommands_via_dirglob
         subcommand_files
       end
 
-      def find_subcommands_via_rubygems
-        files = find_files_latest_gems 'chef/knife/*.rb'
-        subcommand_files = {}
-        files.each do |file|
-          rel_path = file[/(#{Regexp.escape File.join('chef', 'knife', '')}.*)\.rb/, 1]
-
-          # When not installed as a gem (ChefDK/appbundler in particular), AND
-          # a different version of Chef is installed via gems, `files` will
-          # include some files from the 'other' Chef install. If this contains
-          # a knife command that doesn't exist in this version of Chef, we will
-          # get a LoadError later when we try to require it.
-          next if from_different_chef_version?(file)
-
-          subcommand_files[rel_path] = file
-        end
-
-        subcommand_files.merge(find_subcommands_via_dirglob)
-      end
-
-      def have_plugin_manifest?
-        plugin_manifest_path && File.exist?(plugin_manifest_path)
-      end
-
-      def plugin_manifest
-        Chef::JSONCompat.from_json(File.read(plugin_manifest_path))
-      end
-
-      def plugin_manifest_path
-        Chef::Util::PathHelper.home('.chef', 'plugin_manifest.json')
+      #
+      # Subclassses should define this themselves.  Eventually, this will raise a
+      # NotImplemented error, but for now, we mimic the behavior the user was likely
+      # to get in the past.
+      #
+      def subcommand_files
+        Chef::Log.warn "DEPRECATED: Using Chef::Knife::SubcommandLoader directly is deprecated.
```

Comment:
Should probably use `Chef::Log.deprecation` which has the magic where you can turn deprecation warnings into hard errors.